### PR TITLE
EOS-28492: Handle SIGTERM signal for m0d process

### PR DESCRIPTION
**Pull request by VinothVijayakumar-dev**
Passing sigterm signal before sigkill is passed

Signed-off-by: Vinoth.V <vinoth.v@seagate.com>

# Problem Statement
For both sigterm and sigkill, kill command was sent

ADDB data is completely not available when SIGKILL is used.
In a graceful shutdown case too, SIGKILL is used, due to which
not able to collect ADDB data base even for controlled shutdown
case.
To fix this issue use SIGTERM for graceful shutdown cases.

# Coding
   Checklist for Author
-  [yes] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [yes] Unit and System Tests are added
- [yes] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [yes] Testing was performed with RPM

Adding Deployment results
-----------------------------------------------------

> [root@ssc-vm-g4-rhev4-0383 k8_cortx_cloud]# kubectl exec -it cortx-data-ssc-vm-g4-rhev4-0383-7c65ffff74-df4t5 -c cortx-hax -- hctl status
Byte_count:
    critical_byte_count : 0
    damaged_byte_count : 0
    degraded_byte_count : 0
    healthy_byte_count : 0
Data pool:
    # fid name
    0x6f00000000000001:0x33 'storage-set-1__sns'
Profile:
    # fid name: pool(s)
    0x7000000000000001:0x50 'Profile_the_pool': 'storage-set-1__sns' 'storage-set-1__dix' None
Services:
    cortx-server-headless-svc-ssc-vm-g4-rhev4-0383
    [started]  hax        0x7200000000000001:0x2d  inet:tcp:cortx-server-headless-svc-ssc-vm-g4-rhev4-0383@2001
    [started]  rgw        0x7200000000000001:0x30  inet:tcp:cortx-server-headless-svc-ssc-vm-g4-rhev4-0383@5001
    cortx-data-headless-svc-ssc-vm-g4-rhev4-0383  (RC)
    [started]  hax        0x7200000000000001:0x7   inet:tcp:cortx-data-headless-svc-ssc-vm-g4-rhev4-0383@2001
    [started]  ioservice  0x7200000000000001:0xa   inet:tcp:cortx-data-headless-svc-ssc-vm-g4-rhev4-0383@3001
    [started]  ioservice  0x7200000000000001:0x19  inet:tcp:cortx-data-headless-svc-ssc-vm-g4-rhev4-0383@3002
    [started]  confd      0x7200000000000001:0x28  inet:tcp:cortx-data-headless-svc-ssc-vm-g4-rhev4-0383@3003

> [root@ssc-vm-g4-rhev4-0383 k8_cortx_cloud]# kubectl get pods
NAME                                                 READY   STATUS    RESTARTS   AGE
consul-client-p7nhh                                  1/1     Running   0          24h
consul-server-0                                      1/1     Running   0          24h
cortx-control-75d6bdccdb-m6hcd                       1/1     Running   0          24h
cortx-data-ssc-vm-g4-rhev4-0383-7c65ffff74-df4t5     4/4     Running   0          24h
cortx-server-ssc-vm-g4-rhev4-0383-58c48d56d9-cz8kv   2/2     Running   0          24h
kafka-0                                              1/1     Running   0          24h
openldap-0                                           1/1     Running   0          24h
zookeeper-0                                          1/1     Running   0          24h
[root@ssc-vm-g4-rhev4-0383 k8_cortx_cloud]#

### CODE REVIEW

File: scripts/install/usr/libexec/cortx-motr/motr-start [hunk 1]
```diff
@@ -32,6 +32,7 @@ _handle_sigterm() {
    if [[ -z $PID ]]; then
       echo "No m0d process to kill"
    else
+      kill -SIGTERM $PID 2> /dev/null
```
Comment (madhavemuri) [inline scripts/install/usr/libexec/cortx-motr/motr-start:35]: Is there any way to know which Signal is received and pass only that signal here ?

Comment (VinothVijayakumar-dev) [reply scripts/install/usr/libexec/cortx-motr/motr-start]: Added 2 separate functions for each signal handlers

**Comment by VinothVijayakumar-dev** [comment]
retest this please.

**Comment by VinothVijayakumar-dev** [comment]
retest this please

File: scripts/install/opt/seagate/cortx/motr/bin/motr_setup [hunk 2]
```diff
@@ -356,5 +358,5 @@ def main(argv: dict):
         return errno.EINVAL
 
 if __name__ == '__main__':
+    signal.signal(signal.SIGTERM, receiveSigTerm)
     sys.exit(main(sys.argv))
```
Comment (madhavemuri) [inline scripts/install/opt/seagate/cortx/motr/bin/motr_setup:362]: How SIGKILL is handled here ?

Comment (VinothVijayakumar-dev) [reply scripts/install/opt/seagate/cortx/motr/bin/motr_setup:362]: python cannot handle SIGKILL signal. 
https://docs.python.org/3/library/signal.html#signal.SIGKILL)

**Comment by VinothVijayakumar-dev** [comment]
retest this please

**Comment by VinothVijayakumar-dev** [comment]
[Delete pod results](https://seagatetechnology-my.sharepoint.com/:x:/g/personal/vinoth_v_seagate_com/ESg9_WMNNx1EsFYUeUgPzi8Bq-z_4_-oMM9ki80Z26J2PA?email=madhav.vemuri%40seagate.com&e=RcHxw7)

**Review by madhavemuri — APPROVED** [review]

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py [hunk 3]
```diff
@@ -1403,6 +1418,29 @@ def fetch_fid(self, service, idx):
...
+    '''
+    Get list of running m0d process
+    '''
+    listOfProc = []
+    # Iterate over the list
+    for proc in psutil.process_iter():
+       try:
+           # Fetch process details as dict
+           pinfo = proc.as_dict(attrs=['pid', 'name', 'username'])
+           if pinfo.get('name') == "m0d":
```
Comment (mehjoshi) [inline scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py:1431]: Should the name == "m0d" or "lt-m0d" ?

Comment (VinothVijayakumar-dev) [reply scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py:1431]: `[root@cortx-data-headless-svc-ssc-vm-rhev4-2078 /]# ps aux | grep m0d
root         99  0.0  0.0  12268  1656 ?        S    05:36   0:00 /usr/bin/bash /usr/libexec/cortx-motr/motr-start m0d-0x7200000000000001:0xa
root        100  0.2  1.9 14786364 318792 ?     Sl   05:36   0:40 /usr/bin/m0d -e libfab:inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2078@21001 -A linuxstob:/etc/cortx/log/motr/0a75d9db67d245bab0ee37d614317067/addb/m0d-0x7200000000000001:0xa/addb-stobs -f <0x7200000000000001:0xa> -T ad -S stobs -D db -m 524288 -q 16 -H inet:tcp:cortx-data-headless-svc-ssc-vm-rhev4-2078@22001 -U -B /dev/sdg1 -z 9999220736 -r 134217728
`
The process name running in containers is m0d. It looks good.

**Review by atulsdeshmukh2312 — APPROVED** [review]

**Pull request merged**